e2e/test-component/scenarios/embedding-sdk/editable-dashboard.cy.spec.tsx

```diff
@@ -45,4 +45,65 @@ describeEE("scenarios > embedding-sdk > editable-dashboard", () => {
     cy.findByRole("tab", { name: "Overview" }).should("not.exist");
     cy.findByRole("tab", { name: "History" }).should("not.exist");
   });
+
+  describe("loading behavior for both entity IDs and number IDs (metabase#49581)", () => {
+    const successTestCases = [
+      {
+        name: "correct entity ID",
+        dashboardIdAlias: "@dashboardEntityId",
+      },
+      {
+        name: "correct number ID",
+        dashboardIdAlias: "@dashboardId",
+      },
+    ];
+
+    const failureTestCases = [
+      {
+        name: "wrong entity ID",
+        dashboardId: "VFCGVYPVtLzCtt4teeoW4",
+      },
+      {
+        name: "one too many entity ID character",
+        dashboardId: "VFCGVYPVtLzCtt4teeoW49",
+      },
+      {
+        name: "wrong number ID",
+        dashboardId: 9999,
+      },
+    ];
+
+    successTestCases.forEach(({ name, dashboardIdAlias }) => {
+      it(`should load dashboard content for ${name}`, () => {
+        cy.get(dashboardIdAlias).then(dashboardId => {
+          mountSdkContent(<EditableDashboard dashboardId={dashboardId} />);
+        });
+
+        getSdkRoot().within(() => {
+          cy.findByDisplayValue("Embedding SDK Test Dashboard").should(
+            "be.visible",
+          );
+          cy.findByText("This dashboard is looking empty.").should(
+            "be.visible",
+          );
+        });
+      });
+    });
+
+    failureTestCases.forEach(({ name, dashboardId }) => {
+      it(`should show an error message for ${name}`, () => {
```

Comment:
We can also use `it.each` here

Reply:
answered here https://github.com/metabase/metabase/pull/49714#discussion_r1909912019